### PR TITLE
feat: add dependency management and upgrade Makefile

### PR DESCRIPTION
**Pull request by Ian2012**
## Description

This PR adds dependency management and upgrades the Makefile accordingly. And adds a CI workflow to publish to pipy

This opens the door for a pre-commit hook (for formatting).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by Ian2012** [comment]
Added more workflows for tests and to upgrade python dependencies

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Run all tests & checks
+
+on:
+  push:
+    branches: [ main, nightly ]
```
Comment (bmtcril) [inline .github/workflows/test.yml:5]: We don't actually have a nightly branch on these

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip setuptools==44.0.0
```
Comment (bmtcril) [inline .github/workflows/test.yml:22]: Is there a reason why this is pinned here instead of using the pip.txt like we do in other actions?

**Review by bmtcril — COMMENTED** [review]
Just a couple of questions, otherwise looks good

File: .github/workflows/test.yml (hunk 1 above)
Comment (Ian2012) [reply .github/workflows/test.yml]: Removed

File: .github/workflows/test.yml (hunk 2 above)
Comment (Ian2012) [reply .github/workflows/test.yml]: Not really, removed

**Review by bmtcril — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-superset&entry.867055334=https://github.com/openedx/tutor-contrib-superset/pull/22&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-04-18+20:40&entry.1917517419=2023-04-20+16:12&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**